lua/lualine/components/diagnostics/init.lua

```diff
@@ -104,13 +104,18 @@ function M:update_status()
 
   -- format the counts with symbols and highlights
   if self.options.colored then
-    local colors = {}
+    local colors, bgs = {}, {}
     for name, hl in pairs(self.highlight_groups) do
       colors[name] = modules.highlight.component_format_highlight(hl)
+      bgs[name] = modules.utils.extract_highlight_colors(hl.name, 'bg') or nil
```

Comment:
this is incorrect . hl is actually only meant to be consumed by `component_format_highlight` . It knows how to handle all cases regarding the highlight . hl.name is actually internal representation and isn't always same as the actual neovims highlight group name (because in some cases we need to create several highlight groups for different modes from same lualines hl)

It should be something like this
```lua
bgs[name] = modules.utils.extract_highlight_colors(colors[name]:match('%%#(.-)#', 'bg')
```